### ISSUE TITLE
FileExistsMode.REPLACE not working

### ISSUE THREAD

**Issue by haidangbk** [spring-projects/spring-integration#8563]
In version 6.0.3

I'm new to spring integration

FileExistsMode.REPLACE throw exception instead it changes the original file.
I don't know if this is a bug.

![image](https://user-images.githubusercontent.com/46040940/221626185-ef026aac-63ca-40dd-8dfe-7104aff62886.png)

**Comment by artembilan** [comment]
Can we see the whole stack trace, please?

**Comment by haidangbk** [comment]
> Can we see the whole stack trace, please?

This is full stack trace.

org.apache.sshd.sftp.common.SftpException: Failure
	at org.apache.sshd.sftp.client.impl.AbstractSftpClient.throwStatusException(AbstractSftpClient.java:217) ~[sshd-sftp-2.9.2.jar:2.9.2]
	at org.apache.sshd.sftp.client.impl.AbstractSftpClient.checkResponseStatus(AbstractSftpClient.java:212) ~[sshd-sftp-2.9.2.jar:2.9.2]
	at org.apache.sshd.sftp.client.impl.AbstractSftpClient.checkResponseStatus(AbstractSftpClient.java:186) ~[sshd-sftp-2.9.2.jar:2.9.2]
	at org.apache.sshd.sftp.client.impl.AbstractSftpClient.checkCommandStatus(AbstractSftpClient.java:164) ~[sshd-sftp-2.9.2.jar:2.9.2]
	at org.apache.sshd.sftp.client.impl.AbstractSftpClient.rename(AbstractSftpClient.java:647) ~[sshd-sftp-2.9.2.jar:2.9.2]
	at org.apache.sshd.sftp.client.SftpClient.rename(SftpClient.java:617) ~[sshd-sftp-2.9.2.jar:2.9.2]
	at org.springframework.integration.sftp.session.SftpSession.rename(SftpSession.java:162) ~[spring-integration-sftp-6.0.3.jar:6.0.3]
	at org.springframework.integration.file.remote.session.CachingSessionFactory$CachedSession.rename(CachingSessionFactory.java:252) ~[spring-integration-file-6.0.3.jar:6.0.3]
	at org.springframework.integration.file.remote.RemoteFileTemplate.doSend(RemoteFileTemplate.java:606) ~[spring-integration-file-6.0.3.jar:6.0.3]
	at org.springframework.integration.file.remote.RemoteFileTemplate.sendFileToRemoteDirectory(RemoteFileTemplate.java:570) ~[spring-integration-file-6.0.3.jar:6.0.3]
	at org.springframework.integration.file.remote.RemoteFileTemplate.doSend(RemoteFileTemplate.java:353) ~[spring-integration-file-6.0.3.jar:6.0.3]
	at org.springframework.integration.file.remote.RemoteFileTemplate.lambda$send$0(RemoteFileTemplate.java:314) ~[spring-integration-file-6.0.3.jar:6.0.3]
	at org.springframework.integration.file.remote.RemoteFileTemplate.execute(RemoteFileTemplate.java:452) ~[spring-integration-file-6.0.3.jar:6.0.3]
	at org.springframework.integration.file.remote.RemoteFileTemplate.send(RemoteFileTemplate.java:314) ~[spring-integration-file-6.0.3.jar:6.0.3]
	at org.springframework.integration.file.remote.RemoteFileTemplate.send(RemoteFileTemplate.java:302) ~[spring-integration-file-6.0.3.jar:6.0.3]
	at org.springframework.integration.file.remote.RemoteFileTemplate.send(RemoteFileTemplate.java:294) ~[spring-integration-file-6.0.3.jar:6.0.3]
	at org.springframework.integration.file.remote.handler.FileTransferringMessageHandler.handleMessageInternal(FileTransferringMessageHandler.java:207) ~[spring-integration-file-6.0.3.jar:6.0.3]
	at org.springframework.integration.handler.AbstractMessageHandler.doHandleMessage(AbstractMessageHandler.java:105) ~[spring-integration-core-6.0.3.jar:6.0.3]
	at org.springframework.integration.handler.AbstractMessageHandler.handleMessage(AbstractMessageHandler.java:73) ~[spring-integration-core-6.0.3.jar:6.0.3]
	at org.springframework.integration.handler.ReplyProducingMessageHandlerWrapper.handleRequestMessage(ReplyProducingMessageHandlerWrapper.java:59) ~[spring-integration-core-6.0.3.jar:6.0.3]
	at org.springframework.integration.handler.AbstractReplyProducingMessageHandler.handleMessageInternal(AbstractReplyProducingMessageHandler.java:136) ~[spring-integration-core-6.0.3.jar:6.0.3]
	at org.springframework.integration.handler.AbstractMessageHandler.doHandleMessage(AbstractMessageHandler.java:105) ~[spring-integration-core-6.0.3.jar:6.0.3]
	at org.springframework.integration.handler.AbstractMessageHandler.handleMessage(AbstractMessageHandler.java:73) ~[spring-integration-core-6.0.3.jar:6.0.3]
	at org.springframework.integration.dispatcher.AbstractDispatcher.tryOptimizedDispatch(AbstractDispatcher.java:115) ~[spring-integration-core-6.0.3.jar:6.0.3]
	at org.springframework.integration.dispatcher.UnicastingDispatcher.doDispatch(UnicastingDispatcher.java:133) ~[spring-integration-core-6.0.3.jar:6.0.3]
	at org.springframework.integration.dispatcher.UnicastingDispatcher.dispatch(UnicastingDispatcher.java:106) ~[spring-integration-core-6.0.3.jar:6.0.3]
	at org.springframework.integration.channel.AbstractSubscribableChannel.doSend(AbstractSubscribableChannel.java:72) ~[spring-integration-core-6.0.3.jar:6.0.3]
	at org.springframework.integration.channel.AbstractMessageChannel.sendInternal(AbstractMessageChannel.java:373) ~[spring-integration-core-6.0.3.jar:6.0.3]
	at org.springframework.integration.channel.AbstractMessageChannel.send(AbstractMessageChannel.java:327) ~[spring-integration-core-6.0.3.jar:6.0.3]
	at org.springframework.integration.channel.AbstractMessageChannel.send(AbstractMessageChannel.java:297) ~[spring-integration-core-6.0.3.jar:6.0.3]
	at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:187) ~[spring-messaging-6.0.5.jar:6.0.5]
	at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:166) ~[spring-messaging-6.0.5.jar:6.0.5]
	at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:47) ~[spring-messaging-6.0.5.jar:6.0.5]
	at org.springframework.messaging.core.AbstractMessageSendingTemplate.send(AbstractMessageSendingTemplate.java:109) ~[spring-messaging-6.0.5.jar:6.0.5]
	at org.springframework.messaging.core.AbstractMessageSendingTemplate.convertAndSend(AbstractMessageSendingTemplate.java:151) ~[spring-messaging-6.0.5.jar:6.0.5]
	at org.springframework.messaging.core.AbstractMessageSendingTemplate.convertAndSend(AbstractMessageSendingTemplate.java:143) ~[spring-messaging-6.0.5.jar:6.0.5]
	at org.springframework.integration.gateway.MessagingGatewaySupport.send(MessagingGatewaySupport.java:467) ~[spring-integration-core-6.0.3.jar:6.0.3]
	at org.springframework.integration.gateway.GatewayProxyFactoryBean.sendOrSendAndReceive(GatewayProxyFactoryBean.java:669) ~[spring-integration-core-6.0.3.jar:6.0.3]
	at org.springframework.integration.gateway.GatewayProxyFactoryBean.invokeGatewayMethod(GatewayProxyFactoryBean.java:584) ~[spring-integration-core-6.0.3.jar:6.0.3]
	at org.springframework.integration.gateway.GatewayProxyFactoryBean.doInvoke(GatewayProxyFactoryBean.java:550) ~[spring-integration-core-6.0.3.jar:6.0.3]
	at org.springframework.integration.gateway.GatewayProxyFactoryBean.invoke(GatewayProxyFactoryBean.java:540) ~[spring-integration-core-6.0.3.jar:6.0.3]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184) ~[spring-aop-6.0.5.jar:6.0.5]
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:218) ~[spring-aop-6.0.5.jar:6.0.5]
	at jdk.proxy2/jdk.proxy2.$Proxy66.sendToSftp(Unknown Source) ~[na:na]
	at com.example.springintegration.controller.UploadFileController.upload(UploadFileController.java:30) ~[classes/:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:568) ~[na:na]
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:207) ~[spring-web-6.0.5.jar:6.0.5]
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:152) ~[spring-web-6.0.5.jar:6.0.5]
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:117) ~[spring-webmvc-6.0.5.jar:6.0.5]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:884) ~[spring-webmvc-6.0.5.jar:6.0.5]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:797) ~[spring-webmvc-6.0.5.jar:6.0.5]
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87) ~[spring-webmvc-6.0.5.jar:6.0.5]
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1081) ~[spring-webmvc-6.0.5.jar:6.0.5]
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:974) ~[spring-webmvc-6.0.5.jar:6.0.5]
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1011) ~[spring-webmvc-6.0.5.jar:6.0.5]
	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:903) ~[spring-webmvc-6.0.5.jar:6.0.5]
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:705) ~[tomcat-embed-core-10.1.5.jar:6.0]
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:885) ~[spring-webmvc-6.0.5.jar:6.0.5]
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:814) ~[tomcat-embed-core-10.1.5.jar:6.0]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:223) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53) ~[tomcat-embed-websocket-10.1.5.jar:10.1.5]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:185) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100) ~[spring-web-6.0.5.jar:6.0.5]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.0.5.jar:6.0.5]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:185) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93) ~[spring-web-6.0.5.jar:6.0.5]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.0.5.jar:6.0.5]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:185) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201) ~[spring-web-6.0.5.jar:6.0.5]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.0.5.jar:6.0.5]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:185) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:177) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:542) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:119) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:357) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:400) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:859) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1734) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at java.base/java.lang.Thread.run(Thread.java:833) ~[na:na]

**Comment by artembilan** [comment]
So, it fails on `at org.apache.sshd.sftp.client.SftpClient.rename(SftpClient.java:617) ~[sshd-sftp-2.9.2.jar:2.9.2]`.
Would be great to know what is going on the server side: perhaps your user doesn't have permissions for renaming...

**Comment by haidangbk** [comment]
I have set chmod read, write, execute for user in sftp directory.
![image](https://user-images.githubusercontent.com/46040940/221857451-d3de76c6-494d-4a84-8829-dbd39c2d690d.png)

**Comment by artembilan** [comment]
OK. Then this one is out of this project scope.
In fact the `FileExistsMode.REPLACE` is a default one and if it didn't work, everyone would complain.
We have many tests in the project against Apache MINA SFTP server and nothing fails.
I only can suggest to consult with your SFTP server to determine why rename  command is rejected.

**Comment by der-markus** [comment]
Just came along this scenario in my project .

We recently switched to use IntegrationFlow with FileExistsMode.FAIL in the first step.
After a couple of days business came along and requested to set FileExistsMode.REPLACE instead.

As of now we are facing exactly the same issue.

We do have enabled useTemporaryFilename.
- The SFTP Integration is writing the filename.ext.writing file to SFTP server. ✅ 
- After writing is finished the file is rename to filename.ext ✅ 
- We need to write a file with the same filename again.
- The SFTP Integration is writing the filename.ext.writing file to SFTP server. ✅ 
- After writing is finished the file is rename to filename.ext 🛑 

So in the end the old  filename.ext named file is still on SFTP. 
Next to it there is the filename.ext.writing file with the new content.

And the log says:

`Caused by: org.springframework.messaging.MessagingException: Failed to write to '/develop/export/XXXXXXX.json.writing' while uploading the file
at org.springframework.integration.file.remote.RemoteFileTemplate.sendFileToRemoteDirectory(RemoteFileTemplate.java:573)
at org.springframework.integration.file.remote.RemoteFileTemplate.doSend(RemoteFileTemplate.java:353)
... 74 common frames omitted
Caused by: org.apache.sshd.sftp.common.SftpException: Failure
at org.apache.sshd.sftp.client.impl.AbstractSftpClient.throwStatusException(AbstractSftpClient.java:217)
at org.apache.sshd.sftp.client.impl.AbstractSftpClient.checkResponseStatus(AbstractSftpClient.java:212)
at org.apache.sshd.sftp.client.impl.AbstractSftpClient.checkResponseStatus(AbstractSftpClient.java:186)
at org.apache.sshd.sftp.client.impl.AbstractSftpClient.checkCommandStatus(AbstractSftpClient.java:164)
at org.apache.sshd.sftp.client.impl.AbstractSftpClient.rename(AbstractSftpClient.java:647)
at org.apache.sshd.sftp.client.SftpClient.rename(SftpClient.java:617)
at org.springframework.integration.sftp.session.SftpSession.rename(SftpSession.java:165)
at org.springframework.integration.file.remote.session.CachingSessionFactory$CachedSession.rename(CachingSessionFactory.java:252)
at org.springframework.integration.file.remote.RemoteFileTemplate.doSend(RemoteFileTemplate.java:606)
at org.springframework.integration.file.remote.RemoteFileTemplate.sendFileToRemoteDirectory(RemoteFileTemplate.java:570)
... 75 common frames omitted`

We do have all the rights we need on SFTP Server to perform 

Are we missing any configuration? SFTP Documentation tells me to remove the target file before renaming. I thought SFTP Integration would handle this for me?

**Comment by artembilan** [comment]
> at org.springframework.integration.sftp.session.SftpSession.rename(SftpSession.java:165)

The logic there is like this:
```
	@Override
	public void rename(String pathFrom, String pathTo) throws IOException {
		if (this.sftpClient.getVersion() >= SftpConstants.SFTP_V5) {
			this.sftpClient.rename(pathFrom, pathTo, SftpClient.CopyMode.Overwrite);
		}
		else {
			try {
				this.sftpClient.rename(pathFrom, pathTo);
			}
			catch (SftpException sftpex) {
				if (SftpConstants.SSH_FX_FILE_ALREADY_EXISTS == sftpex.getStatus()) {
					remove(pathTo);
					// attempt to rename again
					this.sftpClient.rename(pathFrom, pathTo);
				}
				else {
					throw sftpex;
				}
			}
		}
	}
```
You fail exactly on that `this.sftpClient.rename(pathFrom, pathTo);`, but looks like the status of that failure is not a `SftpConstants.SSH_FX_FILE_ALREADY_EXISTS`, but just general `SSH_FX_FAILURE`.
In the old Spring Integration version `5.5.x` we really just treat all the exceptions as failure and call remove before trying to rename again.

However I don't think it is always legit to remove old file for any exceptions thrown from the server.
We always can use a retry advice on our channel adapter to attempt sending the file again.

Would be great if you can consult with your SFTP server to see why its rename command does not return a proper error code.
Or it would be better to use an SFTP v5 or higher.

As I said before: we see that some error is thrown from rename command, but not all of them justify the file removal.

**Comment by artembilan** [comment]
OK. Repurpose this issue for temporary file removal when rename to the target file has failed.

**Comment by artembilan** [comment]
Yeah... Looks like we cannot remove that tmp file.
We have a logic like this:
```
if (FileExistsMode.REPLACE.equals(mode)) {
	session.write(stream, tempFilePath);
}
else if (FileExistsMode.APPEND.equals(mode)) {
	session.append(stream, tempFilePath);
}
```

so, after removing we may lose an old data we have just appended to.

I think we need to change the logic to something similar what we have with FTP and SMB:
```
this.client.deleteFile(pathTo);
boolean completed = this.client.rename(pathFrom, pathTo);
```
```
if (smbFileTo.exists()) {
	smbFileTo.delete();
}
smbFileFrom.renameTo(smbFileTo);
```

I mean we have to remove the target existing file unconditionally before rename a tmp one into it.

**Comment by dlehammer** [comment]
**TLDR; Fix supporting SFTP protocol v4 or older is included in [v6.2.0+](https://github.com/spring-projects/spring-integration/releases/tag/v6.2.0-M1)**

For posterity, I thought I'd take a stab at outlining how both the initial symptom described by @haidangbk and the [excerpt](https://github.com/spring-projects/spring-integration/issues/8563#issuecomment-1448205545) quoted below from @artembilan can be `true` simultaneously.

> We have many tests in the project against Apache MINA SFTP server and nothing fails.

I was in a situation similar to haidangbk, in my case upgrading from v5.5.20 -> v6.1.9, suddenly when deploying live the SFTP `REPLACE` would fail during renaming ~ but at the same time the local `IntegrationFlow` testsuite succeeded.

This new behaviour was caused by the introduction of a new branch in the [SftpSession#rename(..)](https://github.com/spring-projects/spring-integration/blob/d38e0d8720870e40b1f4266c24911ff584b0dd42/spring-integration-sftp/src/main/java/org/springframework/integration/sftp/session/SftpSession.java#L165) logic.
This caused older SFTP servers with SFTP protocols where the status `SftpConstants.SSH_FX_FILE_ALREADY_EXISTS` (11) isn't supported, to `throw` an exception ~ breaking the default `REPLACE` support compared to earlier versions like v5.5.20.

The Apache MINA SFTP server v2.9.3 supports SFTP protocol v6, unfortunately that limits coverage based on this server to the `sftpClient.getVersion() >= SftpConstants.SFTP_V5` branches.
In my case integrating with older 3rd party OpenSSH SFTP servers live, only SFTP protocol v3 was supported server-side ~ limiting the status interval to [0-8](https://winscp.net/eng/docs/sftp_codes#code_4) and an existing file resulting in status 4 during rename.
Upgrading to v6.2.0 resolved the issue 🙏
